### ISSUE TITLE
Change in ToSelectStatement Behavior

### ISSUE THREAD

**Issue by mjdaly** [ServiceStack/Issues#398]
Upon switching from SS 4.0.40 to 4.0.52, `ToSelectStatement` has changed behaviour, it is a breaking change.

**Example Code**
```
    class Program
    {
        public class ChunkedFileChunk
        {
            [AutoIncrement]
            public int Id { get; set; }

            public long ChunkedFileId { get; set; }
            public int Chunk { get; set; }
        }

        static void Main(string[] args)
        {
            IDbConnectionFactory dbFactory = new OrmLiteConnectionFactory("",MySqlDialect.Provider);

            using (var conn = dbFactory.CreateDbConnection())
            {
                var chunksSql =
                    conn.From<ChunkedFileChunk>()
                        .Where(cfc => cfc.ChunkedFileId == 1L)
                        .OrderBy(cfc => cfc.Chunk)
                        .ToSelectStatement();
            
            }
        }
    }
```

**4.0.40 Output**
```
SELECT `Id`, `ChunkedFileId`, `Chunk`, `InsertedUtc`, `Length`, `Bytes` 
FROM `ChunkedFileChunk`
WHERE (`ChunkedFileId` = 1)
ORDER BY `Chunk`

```
**4.0.52 Output**
```
SELECT `Id`, `ChunkedFileId`, `Chunk`, `InsertedUtc`, `Length`, `Bytes` 
FROM `ChunkedFileChunk`
WHERE (`ChunkedFileId` = @0)
ORDER BY `Chunk`
```

We are using `ToSelectStatement` to generate SQL e.g. `conn.SelectLazy<ChunkedFileChunk>(chunksSql);`

**Comment by mythz** [comment]
That's a result from moving away from in-line SQL, the `ToSelectStatement()` isn't supposed to be used directly, it's meant for use by OrmLite higher-level API's. You can access the DB Params you need for the query with:

```csharp
var q = conn.From<ChunkedFileChunk>()
                 .Where(cfc => cfc.ChunkedFileId == 1L)
                 .OrderBy(cfc => cfc.Chunk)
var dbParams = q.Params;

var sql = q.ToSelectStatement();
```

So I'd recommend using the Params when you execute the SQL as we're moving to parameterized SqlExpressions.

For the time being you should be able to revert to existing behavior with:

```csharp
OrmLiteConfig.UseParameterizeSqlExpressions = false;
```

But we're planning on remove inline SQL behavior in SqlExpression future as we don't want to support multiple parallel implementations with multiple configurable states.

**Comment by mjdaly** [comment]
We are only using ToSelectStatement prior to SelectLazy, because SelectLazy doesn't take an `SqlExpression`, is that something you can add in or will there be another way?

**Comment by mythz** [comment]
Yep no worries added in this commit: https://github.com/ServiceStack/ServiceStack.OrmLite/commit/4dd70e68198f6fa375dbc734a51c9410c9f86472

So you can do:

```csharp
db.SelectLazy(conn.From<ChunkedFileChunk>()
                        .Where(cfc => cfc.ChunkedFileId == 1L)
                        .OrderBy(cfc => cfc.Chunk))
```

This change is available from v4.0.53 that's now [available on MyGet](https://github.com/ServiceStack/ServiceStack/wiki/MyGet)

**Comment by mjdaly** [comment]
Excellent, thanks.
